tests/python/pants_test/engine/test_fs.py

```diff
@@ -597,6 +598,39 @@ def test_caches_downloads(self):
           82
         ))
 
+  def generate_original_digest(self) -> Digest:
+    content = b'dummy content'
+    input_files_content = InputFilesContent((
+      FileContent(path='a.txt', content=content),
+      FileContent(path='b.txt', content=content),
+      FileContent(path='c.txt', content=content),
+      FileContent(path='subdir/a.txt', content=content),
+      FileContent(path='subdir/b.txt', content=content),
+      FileContent(path='subdir2/a.txt', content=content),
+    ))
+    digest = self.request_single_product(Digest, input_files_content)
+    return digest
+
+  def test_empty_snapshot_subset(self) -> None:
+    ss = SnapshotSubset(directory_digest=self.generate_original_digest(),
+        include_files=(),
+        include_dirs=(),
+    )
+    subset_digest = self.request_single_product(Digest, ss)
+    assert subset_digest == EMPTY_DIRECTORY_DIGEST
+
+  def test_snapshot_subset_files_dirs(self) -> None:
+    ss = SnapshotSubset(directory_digest=self.generate_original_digest(),
+        include_files=("a.txt", "c.txt"),
+        include_dirs=('subdir2',),
+    )
+    subset_digest = self.request_single_product(Digest, ss)
+    result_files = self.request_single_product(FilesContent, subset_digest)
```

Comment:
Instead, you can request a `Snapshot` and inspect `snapshot.files` to check that it has all the 3 files you care about. `assert set(snapshot.files) == {'a.txt', 'c.txt', 'subdir2/a.txt'}`.

 This has the added benefit of ensuring that there is nothing else contained - only those 3 files.